src/migrate/index.js

```diff
@@ -196,7 +196,7 @@ export default class Migrator {
     .then(batchNo => {
       return this._waterfallBatch(batchNo, migrations, direction)
     })
-    .then(() => this._freeLock())
```

Comment:
If you're using Bluebird, a passthrough such as `.tap` might be cleaner. Same thing though!
http://bluebirdjs.com/docs/api/tap.html